File: web/src/api/client/appSearch.ts
import type { AppSearchResponse } from '../types'
import { apiUrl } from '../base-url'
import { parseDetailOrThrow } from './_shared'

/**
 * Search everything the application owns — sessions, the dialogue inside
 * them, Coding projects and workspaces, Memory pages, scheduled tasks, agents
 * and skills. Repository contents come from `searchEverywhere` instead, which
 * needs an authorized workspace; this one answers in every mode.
 */
export async function searchApp(
  query: string,
  limit = 40,
  signal?: AbortSignal,
): Promise<AppSearchResponse> {
  const res = await fetch(apiUrl('/team/search-app'), {
    method: 'POST',
    headers: { 'Content-Type': 'application/json', Accept: 'application/json' },
    body: JSON.stringify({ query, limit }),
    signal,
  })
  if (!res.ok) await parseDetailOrThrow(res, 'searchApp')
  return res.json()
}

File: app/api/routes/team/app_search.py
"""Application-wide command-palette search — POST /team/search-app.

Repository search lives in :mod:`app.api.routes.team.search_everywhere` and is
scoped to one authorized workspace. This route covers everything the
application itself owns (sessions, dialogue, projects, workspaces, Memory,
scheduled tasks, agents, skills), so the palette can answer a query in Work
mode and before any workspace is chosen.
"""

from __future__ import annotations

from dataclasses import asdict

from fastapi import APIRouter

from app.api.deps import ReadDbSession
from app.api.schemas.app_search import (
    AppSearchItemResponse,
    AppSearchRequest,
    AppSearchResponse,
)
from app.services.app_search_service import search_app

router = APIRouter(prefix="/search-app")


@router.post("", response_model=AppSearchResponse)
async def search_app_route(
    db: ReadDbSession, body: AppSearchRequest
) -> AppSearchResponse:
    items = await search_app(db, body.query, limit=body.limit)
    return AppSearchResponse(
        items=[AppSearchItemResponse.model_validate(asdict(item)) for item in items]
    )
